### PR TITLE
Campaign stats clicks endpoint

### LINKED ISSUE #719: Campaign stats: clicks endpoint
API endpoint

GET /api/campaigns/{id}/clicks

Response object
```json
{
  "total": 123,
  "collection": [
    {
       "link": "http://example.com?foo=bar",
       "unique_clicks": 123,
       "total_clicks": 456
    },
    ...
  ]
}
```

### PR DESCRIPTION
**Pull request by Gligor23**
# Pull Request Template

## Description

Implement Campaign stats: clicks endpoint

API endpoint

GET `/api/campaigns/{id}/clicks`

Response object
```
{
  "total": 123,
  "collection": [
    {
       "link": "http://example.com?foo=bar",
       "unique_clicks": 123,
       "total_clicks": 456
    },
    ...
  ]
}
```

Fixes #719 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A - testing the endpoint locally in insomnia
- [x] Test B - integration tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: actions/campaigns.go [hunk 1]
```diff
@@ -381,3 +382,24 @@ func GetCampaignOpens(c *gin.Context) {
 		"message": "Id must be an integer",
 	})
 }
+
+func GetCampaignClicks(c *gin.Context) {
+	if id, err := strconv.ParseInt(c.Param("id"), 10, 64); err == nil {
```
Comment (FilipNikolovski) [inline actions/campaigns.go:387]: @Gligor23 it's better to return early on each if statement. Switch them.

**Review by FilipNikolovski — CHANGES_REQUESTED** [review]
Add accompanying tests, also we should discuss the SQL and see how it performs with many records.

**Comment by Gligor23** [comment]
@FilipNikolovski you are right this might become heavy query, maybe we should discus this to be summary table or view.

File: actions/campaigns.go [hunk 2]
```diff
@@ -383,23 +383,26 @@ func GetCampaignOpens(c *gin.Context) {
...
-				ClicksStats: stats,
-			})
-			return
-		}
+func GetCampaignClicksStats(c *gin.Context) {
+	id, err := strconv.ParseInt(c.Param("id"), 10, 64)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Id must be an integer",
+		})
```
Comment (FilipNikolovski) [inline actions/campaigns.go:391]: return here.

**Review by FilipNikolovski — CHANGES_REQUESTED** [review]

File: actions/campaigns.go [hunk 3]
```diff
@@ -381,3 +382,27 @@ func GetCampaignOpens(c *gin.Context) {
...
+			"message": "Campaign clicks not found",
+		})
+		return
+	}
+
+	c.JSON(http.StatusOK, entities.CampaignClicksStats{
+		Total:       int64(len(stats)),
+		ClicksStats: stats,
+	})
+	return
```
Comment (FilipNikolovski) [inline actions/campaigns.go:406]: Omit the return here, staticcheck fails for redundant returns.

**Review by FilipNikolovski — CHANGES_REQUESTED** [review]

File: actions/campaigns.go (hunk 2 above)
Comment (Gligor23) [reply actions/campaigns.go:391]: @FilipNikolovski i am resolving this, i added return here

File: actions/campaigns.go (hunk 3 above)
Comment (Gligor23) [reply actions/campaigns.go]: @FilipNikolovski i removed the return here so i am resolving this too

**Comment by FilipNikolovski** [comment]
@Gligor23 regarding the go.mod and go.sum conflicts, we'll need to look at them together, the rest you can resolve it.

**Review by FilipNikolovski — APPROVED** [review]

**Review by FilipNikolovski — APPROVED** [review]

**Pull request merged**